novawallet/Modules/NetworkDetails/NetworkDetailsInteractor.swift

```diff
@@ -20,31 +20,38 @@ final class NetworkDetailsInteractor {
         connectionFactory: ConnectionFactoryProtocol,
         chainRegistry: ChainRegistryProtocol,
         repository: AnyDataProviderRepository<ChainModel>,
-        operationQueue: OperationQueue,
-        nodeMeasureQueue: OperationQueue
+        storageRequestFactory: StorageRequestFactoryProtocol,
+        operationQueue: OperationQueue
     ) {
         self.chain = chain
         self.connectionFactory = connectionFactory
         self.chainRegistry = chainRegistry
         self.repository = repository
+        self.storageRequestFactory = storageRequestFactory
         self.operationQueue = operationQueue
-        self.nodeMeasureQueue = nodeMeasureQueue
     }
 }
 
 // MARK: NetworkDetailsInteractorInputProtocol
 
 extension NetworkDetailsInteractor: NetworkDetailsInteractorInputProtocol {
     func setup() {
-        connectToChainNodes()
+        connectToNodes(of: chain)
         subscribeChainChanges()
     }
 
     func toggleNetwork() {
         let saveOperation = repository.saveOperation({ [weak self] in
             guard let self else { return [] }
 
-            return [chain.byChanging(enabled: !chain.enabled)]
+            var updatedChain = chain.byChanging(enabled: !chain.enabled)
+
+            if chain.enabled {
+                updatedChain = updatedChain.updatingSelectedNode(with: nil)
+                updatedChain = updatedChain.updatingConnectionMode(for: .manual)
```

Comment:
Didn't get why do we need even update connection mode and selected node when chain enabled? Also is selected node is nil then manual connection doesn't make sense as we don't have a node to connect to. Am I right?

Reply:
From the business logic's perspective, yes, you're right. From the design perspective, there is a matching state of connection mode for a disabled network.

It seems like we should just map the view model in the right way considering the network state and not mess with the connection mode of the actual stored chain.

<img width="287" alt="image" src="https://github.com/novasamatech/nova-wallet-ios/assets/34720457/6c18c731-9980-4fd9-bbe7-774b5d89c1bd">